tests/BaseElementTest.php

```diff
@@ -482,17 +482,31 @@ public function testPreviewLink(string $class, string $elementIdentifier, ?strin
         /** @var BaseElement $element */
         $element = $this->objFromFixture($class, $elementIdentifier);
         $previewLink = $element->PreviewLink();
+        // Remove SubsiteID from preview link in case subsites is installed
+        $rx = '/[?|&]SubsiteID=[0-9]+/';
+        if (preg_match($rx, $previewLink)) {
+            $previewLink = preg_replace($rx, '', $previewLink);
+        }
 
         if ($link) {
-            $regex = '/^' . preg_quote($link, '/') . '[?&]' . preg_quote('ElementalPreview=', '/')
+            // Note that the preview link is suffixed with forward slash when subsites is installed
+            // because of BaseElementSubsites::updatePreviewLink() use of HTTP::setGetVar()
+            // which turns relative urls to absolute urls
+            $regex = '/^\/?' . preg_quote($link, '/') . '[?&]' . preg_quote('ElementalPreview=', '/')
                 .'\d*#' . $element->getAnchor() . '$/';
             $this->assertMatchesRegularExpression($regex, $previewLink);
             // Doesn't try to blindly append query string and anchor - but instead merges intelligently with
             // whatever's already there
             $this->assertSame(1, substr_count($previewLink, '?'));
             $this->assertSame(1, substr_count($previewLink, '#'));
         } else {
-            $this->assertSame($link, $previewLink);
+            if ('/' === $previewLink) {
+                // $previewLink will be suffixed with forward slash when subsites is installed
+                // See comment higher up in this method
+                $this->assertSame('/', $previewLink);
+            } else {
+                $this->assertSame($link, $previewLink);
+            }
```

Comment:
You're effectively doing `$this->assertSame($previewLink, $previewLink);`.... let's just strip off the suffix instead.

```suggestion
            $this->assertSame($link, rtrim($previewLink, '/'));
```
It's safe to leave `$link` as it is here, because it's already proven itself to be falsy.

Reply:
Updated